### DISCUSSION TITLE
sections of level.Chunk do not have Y level

### DISCUSSION THREAD

**Discussion by maxsupermanhd** [Tnze/go-mc#161] [Q&A]
I have save.Chunk structure and I want to get blockstates from it at arbitrary position, looks like there is no way to tel elevation of blocks in sections...

Maybe better question will be how do I go from XYZ to blockstate with save.Chunk?

**Comment by Tnze** [comment]
Do you mean `level.Chunk` or `save.Chunk`?

**Reply by maxsupermanhd** [reply]
I mean specifically level.Chunk, for some reason it exists, want to know for what. Looks like a shortcut to me

**Reply by Tnze** [reply]
I don't understand

**Reply by maxsupermanhd** [reply]
There is a function `ChunkFromSave` that is taking `save.Chunk` and converting it into `level.Chunk`. `level.Chunk` structure has sections array that holds already prepared palette containers that were parsed from `save.Chunk` however those sections don't have Y level defined so it is possible that those sections can shift vertically up and down.
Example: I have 1.17 `save.Chunk`, converting to `level.Chunk` will result in loss of Y level for sections and I no longer know from where blocks start from 0 or from -64

**Reply by Tnze** [reply]
You should always use a slice containing all chunks when working with level.Chunk

**Reply by maxsupermanhd** [reply]
You wanted to say sections? I still don't get purpose of `level.Chunk` without Y levels

**Reply by Tnze** [reply]
Do you know the purpose of Y field in save.Chunk?

**Reply by maxsupermanhd** [reply]
I am talking about *sections* of chunk
